### DISCUSSION TITLE
È possibile creare un sensore temperatura, che faccia riferimento alla temperatura  di un integrazione meteo?

### DISCUSSION THREAD

**Discussion by emiliopalmucci** [indomus/forum#2732] [HUB: Home Assistant]
Buongiorno.....volevo sapere se con home assistent è possibile creare un sensore di temperatura, che faccia riferimento alla temperatura rilevata tramite un integrazione meteo (open-meteo)...

**Comment by snakuzzo** [comment]
Sì...con i template. Leggi [QUI](https://indomus.it/formazione/esercizi-di-domotica-su-home-assistant-1-creare-dei-sensori-template/)

**Comment by emiliopalmucci** [comment]
![Screenshot_20231128_102248](https://github.com/indomus/forum/assets/94483137/c38186e1-80a9-4bb0-93c9-6ccb3c20ba91)

**Reply by emiliopalmucci** [reply]
Sto provando, ma questo sensore, chiamato temperatura esterna, proprio non mi compare.

**Reply by snakuzzo** [reply]
Ma non esiste proprio l'entità o c'è ma non hai il valore corretto?
Hai controllato i log dopo il riavvio?

**Comment by emiliopalmucci** [comment]
Adesso mi compare il sensore chiamato "temperatura esterna2" (dovevo riavviare proprio H.A.), ma quando vado a vedere lo stato, su "strumenti per sviluppatori", lo stato mi dà " unknown" ......

**Reply by snakuzzo** [reply]
perché è sbagliato sicuramente il template. Controlla i log e prova da strumenti sviluppatori.

SMETTILA DI CREARE NUOVI THREAD SU QUESTA DISCUSSIONE. Fai reply sotto l'ultimo messaggio!!!

**Reply by emiliopalmucci** [reply]
![Screenshot_20231128_105747](https://github.com/indomus/forum/assets/94483137/f9ada7ea-b9f2-4df2-995d-2e376391b82f)

**Reply by emiliopalmucci** [reply]
Questo è l'entità che contiene la temperatura che mi interessa

**Reply by Scialla** [reply]
```
temperatura_esterna2:
  friendly_name: "Temperatura Esterna2"
  value_template: "{{ state_attr('weather.casa_2', 'temperature') }}"
  unit_of_measurement: '°C'
```
Però dovresti capirle queste cose ed arrivare a costruirti i template da solo....

studia:

https://www.home-assistant.io/docs/configuration/templating/

**Reply by emiliopalmucci** [reply]
Ciao, ho dato uno sguardo all' articolo....ma per chi è un po' a dieta di queste cose, è praticamente impossibile riuscire a comunicare tramite questo linguaggio, a H.A. che dell'entità wather.casa, mostrami solo la temperatura!  Chiedevo aiuto appunto sul forum.   Credo manchi una stringa che vada a dire a h.a. di prendere solo il dato della temperatura

**Reply by Scialla** [reply]
non manca nulla, il codice che ti ho postato crea un sensore (temperatura_esterna2) che "prende" l'attributo temperatura dall'entità weather.casa_2

usa i "modelli" su HA per provare i template, prova, vai, metti il template `{{ state_attr('weather.casa_2', 'temperature') }}` e vedrai che funziona

Nessuno è nato "imparato", tutti ci abbiamo sbattuto la testa e provato fino a capire come funziona, c'è chi ci ha messo più tempo, chi ce ne ha messo meno, chi ancora ci sbatte la testa. L'importante è capire, provare, imparare, anche perché non ci sarà sempre qualcuno che fa il lavoro al posto tuo e tu ti limiti al copia incolla e, anche se ci fosse, al primo problema non sapresti che fare, e ci sono casi in cui, per capire il problema, bisognerebbe essere davanti alla tastiera al posto tuo, impossibile quindi risolverli da remoto.

Il consiglio è quindi cercare di capire, se fai con calma e senza fretta, non è poi così complicato, non questo almeno ;)

**Reply by emiliopalmucci** [reply]
![Screenshot_20231128_144903_edit_26434364661321](https://github.com/indomus/forum/assets/94483137/2ce708e3-d974-4477-884b-ec5ad1b56116)

**Reply by emiliopalmucci** [reply]
Grazie, come vedi alla fine, con la tua stringa riesco a visualizzare il valore desiderato.   Adesso, è da sistemare i decimali, ma questo per dimostrarti che ci metto un po' del mio, se mi metti un link che spiega come poterlo fare, ci proverò da solo!   Grazie ancora!

**Reply by Scialla** [reply]
il link è sempre lo stesso di prima. Che valore ti aspetti? se vuoi spostare la virgola a sx devi dividere il valore per 10

Ripeto: aiutati con "Stumenti per sviluppatori" l'icona del martelletto sulla sx, poi sul tab "modelli"; lì puoi fare tutte le prove che vuoi e vedi il risultato (a dx) in diretta

**Reply by emiliopalmucci** [reply]
Ok!  Era un errore mio! C'era un '1' di troppo all'inizio del tuo link....che avevo aggiunto io per sbaglio.    Grazie ancora  e buona giornata
